homeassistant/components/wyoming/devices.py

```diff
@@ -0,0 +1,144 @@
+"""Class to manage satellite devices."""
+from __future__ import annotations
+
+from collections.abc import Callable, Iterator
+from dataclasses import dataclass, field
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers import area_registry as ar, device_registry as dr
+
+from .const import DOMAIN
+from .data import WyomingService
+
+
+@dataclass
+class SatelliteDevice:
+    """Class to store device."""
+
+    satellite_id: str
+    device_id: str
+    is_active: bool = False
+    is_enabled: bool = True
+    update_listeners: list[Callable[[SatelliteDevice], None]] = field(
+        default_factory=list
+    )
+
+    @callback
+    def set_is_active(self, active: bool) -> None:
+        """Set active state."""
+        self.is_active = active
+        for listener in self.update_listeners:
+            listener(self)
+
+    @callback
+    def set_is_enabled(self, enabled: bool) -> None:
+        """Set enabled state."""
+        self.is_enabled = enabled
+        for listener in self.update_listeners:
+            listener(self)
+
+    @callback
+    def async_pipeline_changed(self) -> None:
+        """Inform listeners that pipeline selection has changed."""
+        for listener in self.update_listeners:
+            listener(self)
+
+    @callback
+    def async_listen_update(
+        self, listener: Callable[[SatelliteDevice], None]
+    ) -> Callable[[], None]:
+        """Listen for updates."""
+        self.update_listeners.append(listener)
+        return lambda: self.update_listeners.remove(listener)
+
+
+class SatelliteDevices:
+    """Class to store devices."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize satellite devices."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self._new_device_listeners: list[Callable[[SatelliteDevice], None]] = []
+        self.devices: dict[str, SatelliteDevice] = {}
+
+    @callback
+    def async_setup(self) -> None:
+        """Set up devices."""
+        for device in dr.async_entries_for_config_entry(
+            dr.async_get(self.hass), self.config_entry.entry_id
+        ):
+            satellite_id = next(
+                (item[1] for item in device.identifiers if item[0] == DOMAIN), None
+            )
+            if satellite_id is None:
+                continue
+            self.devices[satellite_id] = SatelliteDevice(
+                satellite_id=satellite_id,
+                device_id=device.id,
+            )
+
+        @callback
+        def async_device_removed(ev: Event) -> None:
+            """Handle device removed."""
+            removed_id = ev.data["device_id"]
+            self.devices = {
```

Comment:
Why not pop it?

Reply:
The key is the satellite id, not the device id